src/cli/java/org/commcare/util/screen/QueryScreen.java

```diff
@@ -140,9 +142,15 @@ public void answerPrompts(Hashtable<String, String> answers) {
             if (queryPrompt.isSelectOne() && !answer.isEmpty()) {
                 int choiceIndex = Integer.parseInt(answer);
                 Vector<SelectChoice> selectChoices = queryPrompt.getItemsetBinding().getChoices();
-                answer = selectChoices.get(choiceIndex).getValue();
+                if (choiceIndex < selectChoices.size()) {
+                    answer = selectChoices.get(choiceIndex).getValue();
+                } else {
+                    // answer is no longer a valid choice, so clear it out
+                    answer = "";
+                }
             }
             remoteQuerySessionManager.answerUserPrompt(key, answer);
+            refreshItemSetChoices();
```

Comment:
Checking my understanding: `answer` is initially the index in the old choice list, then `getValue` converts it to a value, then `refreshItemSetChoices` updates the dropdown options and also takes care of making sure that if the value is now at a different index, the index gets updated?

Reply:
@orangejenny That's correct. `refreshItemSetChoices` just updates the dropdown options based on the new `values` set and if the current value is not present in the updated dropdown options, it reset the current value for the prompt to empty. This is [the part](https://github.com/dimagi/formplayer/blob/caseClaim+jls/parent-child-case-search/src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java#L46-L49) that updates the index later when sending the answers to the web apps.

Reply:
ah, I hadn't found that last part. Thank you for explaining.

Reply:
I think either I'm misunderstanding or this isn't working quite right. This is with master checked out, and the "mood" dropdown varies depending on the "rating" dropdown. But changing rating is changing the value of mood - that shouldn't happen, right? Shouldn't it either keep the same value or get cleared?
![UGOYa2fw3B](https://user-images.githubusercontent.com/1486591/105860785-f5113200-5fbb-11eb-8b2b-3f16efd80a69.gif)

Reply:
yeah the mood value should get cleared. Will see if I can fix it.